### PR TITLE
Improve ESM2 pretraining VDR

### PR DESCRIPTION
**Pull request by sichu2023**
To address comments on VDR feedback.
https://docs.google.com/document/d/1JnXzuuA_0ovmN0KcO4sGCRNu86KGClaYd4HY0rQgiYM/edit?tab=t.0

### CODE REVIEW

**Comment by sichu2023** [comment]
./build-ci

File: docs/docs/user-guide/examples/bionemo-esm2/pretrain.md [hunk 1]
```diff
@@ -301,9 +301,46 @@ python scripts/protein/esm2/esm2_pretrain.py \
...
+We benchmark our implementation on the following model sizes.
+
+| Model Size | # Layers | Hidden Size | # Attention Heads | FFN Hidden Size |
+|------------|----------|-------------|-------------------|-----------------|
+| 8M         | 8        | 320         | 20                | 1280            |
+| 650M       | 33       | 1280        | 20                | 5120            |
+| 3B         | 36       | 2560        | 40                | 10240           |
+| 15B        | 48       | 5120        | 40                | 20480           |
+
+In our current benchmark, we recommend the following trainiing and device configurations on A100 80GB GPUs to match with the published 2M token global batch size.
```
[the remark this answers is not in the document]
Comment (sichu2023) [reply docs/docs/user-guide/examples/bionemo-esm2/pretrain.md:331]: As a bibtex reference like on their github page?

[the remark this answers is not in the document]
Comment (pstjohn) [reply docs/docs/user-guide/examples/bionemo-esm2/pretrain.md:331]: Preferably just some footnote with a formatted reference

File: docs/docs/user-guide/examples/bionemo-esm2/pretrain.md [hunk 2]
```diff
@@ -301,9 +301,46 @@ python scripts/protein/esm2/esm2_pretrain.py \
...
+| Model Size | # GPUs | Micro Batch Size | Tensor Model Parallel Size |
+|------------|--------|------------------|----------------------------|
+| 8M         | 32     | 64               | 1                          |
+| 650M       | 64     | 32               | 1                          |
+| 3B         | 128    | 16               | 1                          |
+| 15B        | 2048   | 2                | 2                          |
+
+!!! note "Additional Optimization on Micro Batch Size"
+
+    While optimizing micro batch size further might deviate from the published 2,097,152 tokens global batch size exactly, users should observe additional performance boost. For example, 650M model can be trained on a micro batch size of 48.
```
[the remark this answers is not in the document]
Comment (sichu2023) [reply docs/docs/user-guide/examples/bionemo-esm2/pretrain.md:342]: > All models used 2 million tokens as batch size except the 15B model where we used 3.2 million tokens
> batch size.

On page 28 of the bioarxiv [paper](https://www.biorxiv.org/content/10.1101/2022.07.20.500902v1.full.pdf).

[the remark this answers is not in the document]
Comment (sichu2023) [reply docs/docs/user-guide/examples/bionemo-esm2/pretrain.md:342]: It just reminds me of the 3.2M tokens for 15B model. I will have to increase # devices.

File: docs/docs/user-guide/examples/bionemo-esm2/pretrain.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (tshimko-nv) [reply docs/docs/user-guide/examples/bionemo-esm2/pretrain.md:331]: +1 for ref

File: docs/docs/user-guide/examples/bionemo-esm2/pretrain.md [hunk 3]
```diff
@@ -301,9 +301,48 @@ python scripts/protein/esm2/esm2_pretrain.py \
...
+
+!!! note "Memory Allocation from Distributed Optimizer"
+
+    Distributed optimizer is enabled by default for improved memory allocation. Users might observe that the same micro batch size used on multi-device pretraining results in OOM on a single device. If additional optimization is necessary, we recommend running short benchmark on the same number of devices as in the production run.
+
+[^1]: In reference to Lin, Zeming, et al in the original ESM2 publication - "Evolutionary-scale prediction of atomic-level protein structure with a language model".
```
Comment (sichu2023) [inline docs/docs/user-guide/examples/bionemo-esm2/pretrain.md:348]: @tshimko-nv @pstjohn on the suggestion of a footnote reference.

Comment (sichu2023) [reply docs/docs/user-guide/examples/bionemo-esm2/pretrain.md:348]: Let me know if you prefer an official reference as follows.

> Zeming Lin et al. ,Evolutionary-scale prediction of atomic-level protein structure with a language model.Science379,1123-1130(2023).DOI:[10.1126/science.ade2574](https://doi.org/10.1126/science.ade2574)

Comment (sichu2023) [reply docs/docs/user-guide/examples/bionemo-esm2/pretrain.md]: Nice!

**Review by tshimko-nv — APPROVED** [review]

**Pull request merged**